Source/DafnyCore/AST/Modules/ModuleDefinition.cs

```diff
@@ -170,7 +170,7 @@ public class ModuleDefinition : RangeNode, IDeclarationOrUsage, IAttributeBearin
   public string GetCompileName(DafnyOptions options) {
     if (compileName == null) {
       var externArgs = options.DisallowExterns ? null : Attributes.FindExpressions(this.Attributes, "extern");
-      var nonExternSuffix = (options.Get(CommonOptionBag.AddCompileSuffix) ? "_Compile" : "");
+      var nonExternSuffix = (options.Get(CommonOptionBag.AddCompileSuffix) && Name != "_module" && Name != "_System" ? "_Compile" : "");
```

Comment:
Could be slightly nicer to replace these string literals with a shared variable, like `SystemModuleManager.ModuleName` and `DefaultModuleDefinition.Name`

Reply:
I couldn't get either.

Reply:
I'm not sure what you mean, but you'd have to create those constants since they don't exist yet.

Reply:
I see, I was expecting them to be there.